### PR TITLE
oscapd-cli add task setters

### PR DESCRIPTION
**Pull request by ybznek**
I was trying to add interface to make adding getters/setters easier.

If you have better solution/idea for this problem, please share it.

It should support setters from https://github.com/OpenSCAP/openscap-daemon/issues/16

### CODE REVIEW

**Comment by ybznek** [comment]
It cause duplicities in configuration

```
oscapd-cli task 4 enable
oscapd-cli task 4 set-enabled true
```

**Comment by ybznek** [comment]
oscapd-cli task 1 set-title Task
oscapd-cli task 1 set-enabled true
oscapd-cli task 1 set-target ""
oscapd-cli task 1 set-input ""
oscapd-cli task 1 set-tailoring ""
oscapd-cli task 1 set-profile-id "id"
oscapd-cli task 1 set-online-remediation false
oscapd-cli task 1 set-schedule-not-before "2007-03-04T21:08"
oscapd-cli task 1 set-schedule-repeat-after 5

File: bin/oscapd-cli [hunk 1]
```diff
@@ -30,6 +30,75 @@ import sys
...
+
+    @staticmethod
+    def get_bool(values):
+        Task_accessor._expect_param_len(values, 1)
+        bool_val = Task_accessor._parse_bool(values[0])
+        return [bool_val,]
+
+    @staticmethod
+    def _parse_bool(val):
+        if val in ["1", "true", "yes"]:
```
Comment (mpreisler) [inline bin/oscapd-cli]: Please use `val.lower()` to also catch True, TRUE. We allow these variants elsewhere.

File: bin/oscapd-cli [hunk 2]
```diff
@@ -332,13 +403,39 @@ def confirm(prompt):
...
     return False
 
-
-
 def main():
     parser = argparse.ArgumentParser(
         description="OpenSCAP-Daemon command line interface."
     )
     subparsers = parser.add_subparsers()
+    task_accessor = Task_accessor()
```
Comment (mpreisler) [inline bin/oscapd-cli]: Check out PEP8 and class names, `Task_accessor` is not camelcase - https://www.python.org/dev/peps/pep-0008/#class-names

I find the design a bit strange, `Task_accessor` has some static methods and some regular methods, yet it is for all intents and purposes a singleton initialized only here.

File: bin/oscapd-cli [hunk 3]
```diff
@@ -332,13 +403,39 @@ def confirm(prompt):
...
+
+    dbus_iface = None
+    try:
+        dbus_iface = get_dbus_interface()
```
Comment (mpreisler) [inline bin/oscapd-cli]: Moving `dbus_iface` here changes behavior. Previously it was possible to browse CLI help without the daemon running, this is no longer possible.

If you could avoid breaking this use-case I would really appreciate it.

File: bin/oscapd-cli (hunk 1 above)
Comment (ybznek) [inline bin/oscapd-cli]: good idea, thanks

File: bin/oscapd-cli (hunk 3 above)
Comment (ybznek) [inline bin/oscapd-cli]: Thanks for your feedback. It will be fixed in next commit.

File: bin/oscapd-cli (hunk 2 above)
Comment (ybznek) [inline bin/oscapd-cli]: Ah, thanks.

What do you recommend with the design of class?
There are regular methods which have to work with instance variables. 
Static methods don't work with some instance and there is no reason to give them access to "self". Maybe I can move static methods outside of the class, but now it is on one place at the code.

**Comment by ybznek** [comment]
Thanks, for your feedback. I have done separate commit, but I would like to squash them later

File: bin/oscapd-cli [hunk 4]
```diff
@@ -30,6 +30,78 @@ import sys
 from datetime import datetime
 import os.path
 
+class TaskAccessor:
```
Comment (mpreisler) [inline bin/oscapd-cli]: This is an old-style class, make sure you inherit `object` to use the new-style classes in python2.

File: bin/oscapd-cli [hunk 5]
```diff
@@ -30,6 +30,78 @@ import sys
...
+        self._attributes = dict()
+
+    @staticmethod
+    def get_bool(values):
+        TaskAccessor._expect_param_len(values, 1)
+        bool_val = TaskAccessor._parse_bool(values[0])
+        return [bool_val,]
+
+    @staticmethod
+    def _parse_bool(val):
```
Comment (mpreisler) [inline bin/oscapd-cli]: As we discussed, please refactor this so that we can use a single bool parsing function everywhere.

**Comment by ybznek** [comment]
we use "task_create" and "task set-title"

Maybe using always only '_' or only '-' can be better, what do you think @mpreisler ?

**Comment by mpreisler** [comment]
Agreed, feel free to change task_create to task-create but please keep that change isolated in a separate commit. It can go into this PR.

File: bin/oscapd-cli (hunk 4 above)
Comment (ybznek) [inline bin/oscapd-cli]: Thank you, amended.

**Pull request merged**